### DISCUSSION TITLE
Update ESP32-D0WD-V3  4MB

### DISCUSSION THREAD

**Discussion by Martin0475** [hoylabs/OpenDTU-OnBattery#1162] [Q&A] [answered]
Hallo, 

ist es möglich OpenDTU mit 4MB auf eine aktuelle (development) Version upzudaten?

Würde gerne auch neue Features testen.

Nur weis ich nicht welchen Build ich nehmen soll?

Ich vermute mal opendtu-onbattery-generic_esp32_4mb_no_ota.

Und man muss über USB flashen? (Eine Taste drücken oder gedrückt halten?).

Kann da der Rest angeschlossen bleiben?

OpenDTU ist von NordicSkills DTUshop mit XXL Display und für HM und HMS Serie.

Danke schonmal im vorraus.

PS: Wann ist den ein neues offizielles Release geplant?

**Comment by schlimmchen** [comment]
schön, dass du dich für die neuen features und verbesserungen interessierst! korrekt, du kannst die ...4mb_no_ota auf deinem esp32 mit 4mb flash installieren per kabel. wie das geht steht z.b. in der doku des upstream projekts: https://www.opendtu.solar/firmware/flash_esp/#flash-esp32_1

zu deinem p.s.: never ask for an eta 😉 ich arbeite darauf hin, dass es ein neues release gibt. es gab aber einige hinderungsgründe, insb. die 4mb flash thematik. es muss insb. ein gescheites dokument geben, dass den update-vorgang beschreibt. und das würde ich gerne in die neue doku einpflegen, also muss die erst noch präsentierbar werden. ich würde das sehr gerne bis ende nächster woche schaffen, aber das hängt von vielen dingen ab.

**Comment by spcqike** [comment, marked as answer]
> Ich vermute mal opendtu-onbattery-generic_esp32_4mb_no_ota.

Korrekt

> Und man muss über USB flashen? (Eine Taste drücken oder gedrückt halten?).

Auch korrekt. 
das mit der Taste kommt darauf an. Ist in deiner DTU ein Developer Board verbaut? Oder ist es ein eigenes PCB mit verlötetem esp32? Beim dev Board kannst du es wohl einfach Flashen. Bei nem eigenen PCB wieder …. Kommt darauf an. Probiere es aus und berichte :) im Zweifel GPIO_0 low ziehen beim booten des Moduls. 

> Wann ist den ein neues offizielles Release geplant?

gute frage 😄  @schlimmchen @helgeerbe gibt es da Kriterien ab wann ein neues Release eben ein neues Release wird?

Edit: da kamst du mir wohl etwas zuvor (:

**Reply by Martin0475** [reply]
Ich konnte das neue Release mit dem Espressif-Flash-Tool installieren.
Es funktionierte ohne öffnen des Gerätes, d.h. es musste zum flashen kein Knopf gedrückt werden.
Bin am einrichten und testen, Danke an alle.

**Reply by speddy2222** [reply]
Guten Morgen Martin, 
Kannst du mir bitte helfen. Ich stehe vor dem selben Problem.
Ich komme von 2024.06.03 und möchte gerne mein ESP-DOWD-V3 Aktualisieren.
Mit dem Flashtool komme ich nicht weiter. Mein ESP hing in einer Bootschleife fest und nur ein Downgrade auf die 2024.06..03 hat ihm wieder leben eingehaucht.
Kannst du mir bitte sagen wie du ihn aktualisiert hast. Vielen Dank für eure Hilfe
Grüße Tino

**Reply by Snusme83** [reply]
@ speddy2222: Hatte auch immer Probleme beim Flashen der Microcontroller, bis ich rausgefunden hab, dass ich keine Probleme hab, wenn am Microcontroller keine Pins mehr angeschlossen sind. Vielleicht hilft das bei dir auch?

**Reply by Martin0475** [reply]
Hallo Tino,
ich bin nach dieser Anleitung vorgegangen:

https://www.opendtu.solar/firmware/flash_esp/

auf der Seite unten dann "ESP Flash Tools" auswählen und das Programm vom Hersteller herunterladen,
die Haken so setzen wie auf dem Bild, beim ersten mal die Factory.bin verwenden.

Ich habe noch vor dem flashen auf erase gedrückt weil ich komplett neu einrichten wollte.

Hoffe es klappt nun auch bei dir.
Gruß Martin

**Reply by speddy2222** [reply]
Vielen Dank,
das Flashen mit dem Bild und dem ESP Flash Tool hat funktioniert. Nach dem Neustart hat der ESP ein AP geöffnet nur das Standart Kennwort openDTU42 funktioniert nicht  und ich komme nicht auf den Webbrowser.
Gibt es ein neues Passwort?

**Reply by speddy2222** [reply]
Juhu,

Vielen dank Nochmals.

Es hat Funktioniert, keine Ahnung wo der Fehler lag. Nach dem Frühstück nochmal geflasht und sie da. Ich habe die neuste Version auf meinem ESP.
Vielleicht lag es an den leckeren Brötchen :)
Grüße Tino

**Reply by Snusme83** [reply]
🥳 glaub aber nicht, dass es die Brötchen waren. Hast du zum Flashen irgendwelche GPIO abgesteckt?

**Reply by speddy2222** [reply]
Alles steckt in einem 3d Gehäuse mit Siegel an der Schraube wegen der Garantie. Der DTU in von Nordic Skills wie Martin seiner und den will ich nicht öffnen, wie bereits erwähnt, wegen der Garantie.
Ich habe nichts gesteckt oder irgendwelche Tasten gedrückt sondern einfach die neue Version drüber geflasht laut Anleitung.
Martin hatte den richtigen Riecher, sein Tipp mit dem FlashTool war Richtig
.

**Reply by Snusme83** [reply]
Achso, jetzt wird ein Schuh daraus. Du hast die OpenDTUonBattery auf deiner Hardware ohne Speicher oder Laderegler zu haben?🤔

**Reply by speddy2222** [reply]
Nee, hab ich nie gesagt. 
Ich steuere meine Hoymiles mit der DTU on Battery habe ein Shelly Pro 3 EM zur Leistungsmessung und Lade die Batterien mit  2 Victron SmarSolar Laderegler. 
Verbunde habe ich alles mit Venus OS und Iobroker über W-Lan

**Reply by Snusme83** [reply]
Alles klar. Venus OS hatte ich vorher auch.
Dank OpenDTUonBattery kann ich mir dieses jetzt sparen und es läuft alles über einen Mikrocontroller - besser geht es nicht!😎

**Reply by speddy2222** [reply]
kannst du mir kurz deinen Aufbau erklären, vielleicht kann ich meine Anlage damit verbessern?

**Reply by Snusme83** [reply]
6 Module a 370 Wp gehen auf 3 Victron SmartSolar 100/20, diese auf eine Busbar, wo:
1 x Hoymiles HM 1500,
2 x Pylontech US3000C,
1 x Spannungswandler (48 V, 2000 W für Notfall)
1 x Huawei R4850G2 (Laden der Speicher mit Notstromaggregat für Notfall) hängen.

OpenDTUonBattery ist verbunden mit dem HM-1500, Display SH1306, 3 x Smartsolar, 1 x Pylontech. Das war's schon.😉

**Reply by Snusme83** [reply]
![IMG_2572](https://github.com/user-attachments/assets/9652bb71-9043-4762-955d-c20cb57612f8)

**Reply by speddy2222** [reply]
Hast du die Original Victron Busbar, ich finde deine Anlage sehr Interessant.
Ich glaube das ist meine Lösung für eine komplette Autonome Steuerung.
Das bedeutet du schließt alles an der Busbar an, auch der ESP über Gpio, richtig?

Ach quatsch, eine Busbar ist ein Stromverteiler. Wie hast du alles zusammen Vernetzt?

**Reply by Snusme83** [reply]
Nö, nicht original von Victron. Sowas:
![IMG_2573](https://github.com/user-attachments/assets/8dd56e9a-9520-4906-b565-2aba0259b84d)
Die o. g. Sachen sind an der Busbar angeschlossen. Der ESP32- S3 hängt ganz normal an der Haussteckdose, aber stimmt, den müsste ich auch noch irgendwie autark mit Strom versorgen.

**Reply by speddy2222** [reply]
hast du im Netz irgendwo eine Anleitung wie du die ganzen Victron Geräte in die OpenDTU integriert hast?
Ich will das auch wie du.

**Reply by Snusme83** [reply]
Findest du eigentlich alles hier auf dieser tollen Plattform!😉
Kann dir beschreiben, wie ich es gemacht hab, aber schlimmchen darf das nicht lesen, sonst gibt es Schimpfe!😳
Aber im Wiki/ Hardware/ VE.Direct steht es eigentlich.
Du brauchst diese 4-Pin-Jst- Stecker für Ve.direct vom SmartSolar:
![Uploading IMG_2579.jpeg…]()

Diese müssen dann verbunden werden mit den GPIO's des Mikrocontrollers. (Da musst aber an den Microcontroller ran)
Zwecks der galvanischen Trennung müsste aber auch noch so ein Adium???... dazwischen!

Anschlüsse des SmartSolar kannst mit einem Multimeter messen (VCC und GND), dann weißt auch wie bei deinem Laderegler die Belegung ist und kannst dementsprechend die Pins im Pinmapping belegen.

Das ist es mal so grob, wenn es Fragen gibt, einfach melden.

**Reply by Snusme83** [reply]
![IMG_2579](https://github.com/user-attachments/assets/afdbe854-66fa-4100-8fee-11115a8e5b79)

**Reply by Snusme83** [reply]
Hab auch angefangen mit hauptsächlich der Remot Console (Venus OS), dann aber nach und nach die Endgeräte versucht in OpenDTUonBattery zu integrieren und was soll ich sagen, es hat sich gelohnt!

OpenDTUonBattery ist ne absolut runde Sache, alles läuft vollautomatisch, bei Bedarf kann man an allen Stellschrauben drehen, sodass es jeder individuell für seinen Bedarf oder Jahreszeit einstellen kann.

Hab angefangen mit einem normalen BKW, hab gewusst, was ich eigentlich möchte, hatte aber massive Zweifel, dass ich als Laie das schaffe, aber dank der hilfsbereiten Community hier kein Problem - vorausgesetzt man investiert selber Zeit in die Thematik! Aber ich denke das Ergebnis spricht für sich:
![IMG_2582](https://github.com/user-attachments/assets/3809b580-a130-4f75-a624-c8c3c8e17516)

**Comment by speddy2222** [comment]
Ohne hier alles in die Länge zu ziehen und das eigentliche Posting abzuändern. Kann ich dir hier irgendwie direkt schreiben?

**Reply by Snusme83** [reply]
Wüsste nicht wie!

**Reply by speddy2222** [reply]
Wenn du einverstanden bist. Ich würde nur gerne wissen wie du deine 3 Laderegler an einen ESP angeschlossen hast.
Nur wenn es okay ist.
kontakt-huepft-0l@icloud.com
